integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java

```diff
@@ -0,0 +1,145 @@
+            //verify type
+            logger.info("Verifying domain event type {0}", type);
+            assertTrue(event.getType().equals(type));
+            return true;
+          }
+        }
+      } catch (ApiException ex) {
+        Logger.getLogger(ItKubernetesEvents.class.getName()).log(Level.SEVERE, null, ex);
+      }
+      return false;
+    };
+  }
+
+  // Verify the operator instance details are correct
+  private static void verifyOperatorDetails(V1Event event, String opNamespace, String domainUid) throws ApiException {
+    logger.info("Verifying operator details");
+    String operatorPodName = TestActions.getOperatorPodName(OPERATOR_RELEASE_NAME, opNamespace);
+    //verify DOMAIN_API_VERSION
+    assertTrue(event.getInvolvedObject().getApiVersion().equals(TestConstants.DOMAIN_API_VERSION),
+        "Expected " + TestConstants.DOMAIN_API_VERSION + " ,Got " + event.getInvolvedObject().getApiVersion());
+    //verify reporting component to be operator release
+    assertTrue(event.getReportingComponent().equals("weblogic.operator"),
+        "Didn't get reporting component as " + "weblogic.operator");
+    //verify reporting instance to be operator instance
+    assertTrue(event.getReportingInstance().equals(operatorPodName),
+        "Didn't get reporting instance as " + operatorPodName);
+    //verify the event was created by operator
+    Map<String, String> labels = event.getMetadata().getLabels();
+    assertTrue(labels.containsKey("weblogic.createdByOperator")
+        && labels.get("weblogic.createdByOperator").equals("true"));
+    //verify the domainUID matches
+    assertTrue(labels.containsKey("weblogic.domainUID")
+        && labels.get("weblogic.domainUID").equals(domainUid));
+  }
+
+  /**
+   * Returns the description for different event types.
+   *
+   * @param reason event to check for Created, Changed, deleted, processing etc
+   * @param domainUid UID of the domain
+   * @return String containing the complete description
+   */
+  public static String getDomainEventMessage(String reason, String domainUid) {
+    String message = "";
+    switch (reason) {
+      case DOMAIN_CREATED:
+        message = "Domain resource " + domainUid + " was created";
+        break;
+      case DOMAIN_DELETED:
+        message = "Domain resource " + domainUid + " was deleted";
+        break;
+      case DOMAIN_CHANGED:
+        message = "Domain resource " + domainUid + " was changed";
+        break;
+      case DOMAIN_PROCESSING_STARTING:
+        message = "Creating or updating Kubernetes presence for WebLogic Domain with UID " + domainUid;
+        break;
+      case DOMAIN_PROCESSING_COMPLETED:
+        message = "Successfully completed processing domain resource " + domainUid;
+        break;
+      case DOMAIN_PROCESSING_FAILED:
+        message = "Failed to process domain resource " + domainUid
+            + " due to xxxx, the processing  will be retried if required";
```

Comment:
The xxxx in the message needs to be replace with the real error message.

Reply:
removed